### ISSUE TITLE
[no-redeclare] error for declare namespace

### ISSUE THREAD

**Issue by ficristo** [bradzacher/eslint-plugin-typescript#267]
**Repro**
See below.

```JSON
{
  "rules": {
    "no-redeclare": "error"
  }
}
```

```TS
export = CodeMirror;
export as namespace CodeMirror;

declare function CodeMirror(host: HTMLElement, options?: CodeMirror.EditorConfiguration): CodeMirror.Editor;

declare namespace CodeMirror {
}
```

**Expected Result**
No error.

**Actual Result**
`no-redeclare` error for `declare namespace CodeMirror` line.

**Additional Info**

**Versions**

| package                    | version |
| -------------------------- | ------- |
| `eslint-plugin-typescript` | `1.0.0-rc.0` |
| `typescript-eslint-parser` | `eslint-plugin-typescript/parser` |
| `typescript`               | `3.1.6` |

**Comment by bradzacher** [comment]
Investigating this case, it's hard for us to deal with this because it's valid typescript.

TS augments the the base named thing with the namespace, so in this case, it is completely valid code.
The question is; should this should be caught by our linter if it's completely valid code?

Concrete code:
```TS
function CodeMirror(arg: string): boolean {
    return true;
}

namespace CodeMirror {
    export const foo = 1
}

CodeMirror.foo // type === number
CodeMirror("hi") // valid
```

Declarations only:
```TS
declare function CodeMirror(arg : string): boolean;

declare namespace CodeMirror {
    export const foo = 1
}

CodeMirror.foo // type === number
CodeMirror("hi") // valid
```
[repl](https://www.typescriptlang.org/play/index.html#src=%2F%2F%20concrete%20code%20case%0D%0Afunction%20CodeMirror(arg%3A%20string)%3A%20boolean%20%7B%0D%0A%20%20%20%20return%20true%3B%0D%0A%7D%0D%0A%0D%0Anamespace%20CodeMirror%20%7B%0D%0A%20%20%20%20export%20const%20foo%20%3D%201%0D%0A%7D%0D%0A%0D%0ACodeMirror.foo%20%2F%2F%20type%20%3D%3D%3D%20number%0D%0ACodeMirror(%22hi%22)%20%2F%2F%20valid%0D%0A%0D%0A%0D%0A%2F%2F%20declare%20only%20case%0D%0Adeclare%20function%20Foo(arg%20%3A%20string)%3A%20boolean%3B%0D%0A%0D%0Adeclare%20namespace%20Foo%20%7B%0D%0A%20%20%20%20export%20const%20foo%20%3D%201%0D%0A%7D%0D%0A%0D%0AFoo.foo%20%2F%2F%20type%20%3D%3D%3D%20number%0D%0AFoo(%22hi%22)%20%2F%2F%20valid%0D%0A%0D%0A)

**Comment by armano2** [comment]
@bradzacher right now parser is not providing consistent informations about declarations, than we can't really do much for now here.
